### PR TITLE
docs: Custom resource actions

### PR DESCRIPTION
**Pull request by moensch**
Documents how to add custom resource actions to `argocd-cm`.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: docs/operator-manual/resource_actions.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Resource Actions
+
+## Overview
+Argo CD allows operators to define custom actions users can perform on specific resource types. This is used internally to provide actions like `restart` for a `DaemonSet`, or `retry` for an Argo Rollout.
```
Comment (sbose78) [inline docs/operator-manual/resource_actions.md:4]: *custom actions which users can perform

File: docs/operator-manual/resource_actions.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          if obj.spec.suspend ~= nil and obj.spec.suspend then
+              obj.spec.suspend = false
+          end
+
+          return obj
+```
+
+The `discovery.lua` script must return a table where the key name represents the action name. You can optionally include logic to enable or disable certain actions based on the current object state.
+
+The `obj` is a global variable which contains the resource. Each action script must return an optionally modified version of the resource. In this example, we are simply setting `.spec.suspend` to either `true` or false`.
```
Comment (sbose78) [inline docs/operator-manual/resource_actions.md:59]: `false`

File: docs/operator-manual/resource_actions.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            actions["resume"]["disabled"] = false
+        else
+            actions["suspend"]["disabled"] = false
+        end
+
+        return actions
+      definitions:
+      - name: suspend
+        action.lua: |
+          obj.spec.suspend = true
```
Comment (sbose78) [inline docs/operator-manual/resource_actions.md:45]: Would you be able to help us with a more realistic example, please?

**Review by sbose78 — CHANGES_REQUESTED** [review]
Great write-up btw!

**Review by alexmt — CHANGES_REQUESTED** [review]
Thank you for working on custom actions documentation @moensch ! It is a shame we did not have it for such a long time.

Can you please add ` docs/operator-manual/resource_actions.md` to https://github.com/argoproj/argo-cd/blob/master/mkdocs.yml#L52 ?

File: docs/operator-manual/resource_actions.md (hunk 3 above)
Comment (alexmt) [reply docs/operator-manual/resource_actions.md:45]: I think this example is very realistic. It makes it really easy to suspend and resume CronJobs from UI. I would even bundle it into Argo CD .

**Review by alexmt — APPROVED** [review]

**Pull request merged**